src/SqlPersistence/Saga/SqlSaga.cs

```diff
@@ -62,19 +59,37 @@ protected override void ConfigureHowToFindSaga(IConfigureHowToFindSagaWithMessag
             VerifyNoConfigureHowToFindSaga();
             var propertyMapper = new PropertyMapper<TSagaData>(mapper, GetExpression(), GetType());
             ConfigureMapping(propertyMapper);
+
+            if (mapper is IConfigureHowToFindSagaWithFinder finderMapper)
+            {
+                ConfigureFinderMapping(finderMapper);
+            }
+            else
+            {
+                throw new InvalidOperationException($"{mapper.GetType().FullName} must implement {nameof(IConfigureHowToFindSagaWithFinder)}.");
+            }
         }
 
         /// <summary>
         /// Allows messages to be mapped to <see cref="CorrelationPropertyName"/>.
         /// </summary>
         protected abstract void ConfigureMapping(IMessagePropertyMapper mapper);
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="mapper"></param>
+        protected virtual void ConfigureFinderMapping(IConfigureHowToFindSagaWithFinder mapper)
```

Comment:
Not great to have to reimplement core apis like this. We will have to do the same for the new ISagaNotFoundHandlers